test/multicast-dns.spec.js

```diff
@@ -100,14 +110,52 @@ describe('MulticastDNS', () => {
       mdnsA.once('peer', (peerInfo) => {
         expect(pC.id.toB58String()).to.eql(peerInfo.id.toB58String())
         expect(peerInfo.multiaddrs.size).to.equal(1)
-        done()
+        parallel([
+          (cb) => mdnsA.stop(cb),
+          (cb) => mdnsC.stop(cb),
+          (cb) => mdnsD.stop(cb)
+        ], done)
       })
 
       mdnsC.once('peer', (peerInfo) => {})
     })
   })
 
-  it('doesn\'t emit peers after stop', function (done) {
+  it('announces IP6 addresses', function (done) {
+    this.timeout(40 * 1000)
+
+    const options = {
+      port: 50001 // port must be the same
+    }
+    const mdnsA = new MulticastDNS(pA, {
+      broadcast: false, // do not talk to ourself
+      port: 50001
+    })
+    const mdnsB = new MulticastDNS(pB, options)
+
+    parallel([
+      (cb) => mdnsA.start(cb),
+      (cb) => mdnsB.start(cb)
+    ], () => {
+      mdnsA.once('peer', (peerInfo) => {
+        expect(pB.id.toB58String()).to.eql(peerInfo.id.toB58String())
+        expect(peerInfo.multiaddrs.size).to.equal(2)
+        parallel([
+          (cb) => mdnsA.stop(cb),
+          (cb) => mdnsB.stop(cb)
+        ], done)
+      })
+
+      mdnsB.once('peer', (peerInfo) => {})
+    })
+  })
+
+  /*
+   * This test is WRONG.  Peers are async.  How can you assume
+   * that the mdnsA.stop happens before mdnsC gets a response
+   * from mdnsA?
+   */
```

Comment:
@richardschneider wanna fix the test then?

Reply:
I thought about it and could not come up with a test.  How does one prove that something will not happen?

Reply:
You can stop and wait "two intervals" to make sure that indeed no more broadcasts happen.

Reply:
Done